src/azure-cli/azure/cli/command_modules/cdn/_params.py

```diff
@@ -154,3 +138,60 @@ def load_arguments(self, _):
     # Origin #
     with self.argument_context('cdn origin') as c:
         c.argument('origin_name', name_arg_type, id_part='name')
+
+    # WAF #
+
+    with self.argument_context('cdn waf policy set') as c:
+        c.argument('disabled', arg_type=get_three_state_flag())
+        c.argument('block_response_status_code', type=int)
+        c.argument('name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy show') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy delete') as c:
+        c.argument('policy_name', name_arg_type, id_part='name', help='The name of the CDN WAF policy.')
+
+    with self.argument_context('cdn waf policy managed-rule-set') as c:
+        c.argument('policy_name', id_part='name', help='Name of the CDN WAF policy.')
+        c.argument('rule_set_type', help='The type of the managed rule set.')
+        c.argument('rule_set_version', help='The version of the managed rule set.')
+    with self.argument_context('cdn waf policy managed-rule-set list') as c:
+        # List commands cannot use --ids flag
+        c.argument('policy_name', id_part=None)
+    with self.argument_context('cdn waf policy managed-rule-set add') as c:
+        c.argument('enabled', arg_type=get_three_state_flag())
+
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override') as c:
+        c.argument('name', name_arg_type, id_part=None, help='The name of the rule group.')
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override list') as c:
+        # List commands cannot use --ids flag
+        c.argument('policy_name', id_part=None)
+    with self.argument_context('cdn waf policy managed-rule-set rule-group-override set') as c:
+        c.argument('rule_overrides',
+                   options_list=['-r', '--rule-override'],
+                   action=ManagedRuleOverrideAction,
+                   nargs='+')
+
+    with self.argument_context('cdn waf policy custom-rule') as c:
+        c.argument('name', name_arg_type, id_part=None, help='The name of the custom rule.')
+        c.argument('policy_name', id_part='name', help='Name of the CDN WAF policy.')
+    with self.argument_context('cdn waf policy custom-rule list') as c:
+        # List commands cannot use --ids flag
+        c.argument('policy_name', id_part=None)
```

Comment:
It looks the id_part here doesn't work. You don't need to add `id_part` kwargs here.